### PR TITLE
Fix: Recent decision to make FEATURES=pkgdir-index-trusted is now on by default breaks eclean-invalids feature.

### PR DESCRIPTION
**Pull request by siddhanthrathod**
See_also: https://github.com/gentoo/portage/commit/3bc7bfef51d88fd716e882d3931f2873e3493e01

### CODE REVIEW

File: pym/gentoolkit/eclean/search.py [hunk 1]
```diff
@@ -568,7 +567,12 @@ def findPackages(
     dead_binpkgs: dict[str, list[str]] = {}
     keep_binpkgs = {}
 
-    bin_dbapi = portage.binarytree(pkgdir=pkgdir, settings=var_dbapi.settings).dbapi
+    # FEATURES=pkgdir-index-trusted is now on by default which makes portages inavlids
+    # inaccessible
+    settings = var_dbapi.settings
+    settings.unlock()
+    settings.features.remove("pkgdir-index-trusted")
+    bin_dbapi = portage.binarytree(pkgdir=pkgdir, settings=settings).dbapi
```
[the remark this answers is not in the document]
Comment (siddhanthrathod) [reply pym/gentoolkit/eclean/search.py:575]: Yes I'll do this

[the remark this answers is not in the document]
Comment (zmedico) [reply pym/gentoolkit/eclean/search.py:573]: Hmm, this parameter was added in https://github.com/gentoo/portage/commit/c9fb98b4913b286ed404d1df408e51d4f17b1dfb which is not in a portage release yet.

[the remark this answers is not in the document]
Comment (siddhanthrathod) [reply pym/gentoolkit/eclean/search.py:573]: > Hmm, this parameter was added in [gentoo/portage@c9fb98b](https://github.com/gentoo/portage/commit/c9fb98b4913b286ed404d1df408e51d4f17b1dfb) which is not in a portage release yet.

Ah so there might be a chance that portage wont go forward with that change ?

[the remark this answers is not in the document]
Comment (zmedico) [reply pym/gentoolkit/eclean/search.py:573]: Oh, it probably will, but the parameter can't be used unless it's supported. We can used inspect.signature to check if it's supported:

https://docs.python.org/3/library/inspect.html#inspect.signature

[the remark this answers is not in the document]
Comment (siddhanthrathod) [reply pym/gentoolkit/eclean/search.py:574]: This what you meant right ?

[the remark this answers is not in the document]
Comment (siddhanthrathod) [reply pym/gentoolkit/eclean/search.py:574]: > Oh, it probably will, but the parameter can't be used unless it's supported. We can used inspect.signature to check if it's supported:
> 
> https://docs.python.org/3/library/inspect.html#inspect.signature

Ah sorry for earlier that was wrong this works but I'm not so sure if you meant this

[the remark this answers is not in the document]
Comment (zmedico) [reply pym/gentoolkit/eclean/search.py:574]: Looks good! Thanks!

**Review by zmedico — APPROVED** [review]

**Comment by thesamesam** [comment]
Thank you and thank you Zac for reviewing!

**Pull request closed without merging**